src/App.js

```diff
@@ -11,34 +13,181 @@ class App extends React.Component {
       cardDeck: makeShuffledDeck(),
       // currCards holds the cards from the current round
       currCards: [],
+      score: [0, 0],
+      roundsLeft: 26,
+      winner: "undecided",
+      gameEnded: false,
+      gamesWon: [0, 0],
     };
   }
 
+  resetGame = () => {
+    this.setState((state) => {
+      return {
+        cardDeck: makeShuffledDeck(),
+        // currCards holds the cards from the current round
+        currCards: [],
+        score: [0, 0],
+        roundsLeft: 26,
+        winner: "undecided",
+        gameEnded: false,
+      };
+    });
+  };
+
+  computeOutcome = () => {
+    this.setState((state) => {
+      let cards = state.currCards;
+      let currentScore = this.state.score.slice(0);
+      let newScore = this.computeTurnWinner(cards, currentScore);
+      let decision = this.decideWinner(newScore);
+      console.log("computeOutcome function:" + decision);
+      let currentGamesWon = this.state.gamesWon.slice(0);
+      let updatedGamesWon = this.computeGamesWon(decision, currentGamesWon);
+      console.log("computeOutcome() gamesWon variable:" + updatedGamesWon);
+
+      return {
+        // Remove last 2 cards from cardDeck
+        cardDeck: [],
+        // Deal last 2 cards to currCards
+        currCards: [],
+        score: newScore,
+        winner: decision,
+        gameEnded: true,
+        gamesWon: updatedGamesWon,
+      };
+    });
+  };
+
   dealCards = () => {
-    this.setState((state) => ({
-      // Remove last 2 cards from cardDeck
-      cardDeck: state.cardDeck.slice(0, -2),
-      // Deal last 2 cards to currCards
-      currCards: state.cardDeck.slice(-2),
-    }));
+    this.setState((state) => {
+      let cards = state.cardDeck.slice(-2);
+      let currentScore = this.state.score.slice(0);
+      let newScore = this.computeTurnWinner(cards, currentScore);
+      console.log(`dealCards() state variable gamesWon: ${state.gamesWon}`);
+
+      return {
+        // Remove last 2 cards from cardDeck
+        cardDeck: this.state.roundsLeft > 1 ? state.cardDeck.slice(0, -2) : [],
+        // Deal last 2 cards to currCards
+        currCards: state.cardDeck.slice(-2),
+        score: newScore,
+        roundsLeft: state.roundsLeft - 1,
+      };
+    });
+
+    console.log(`card deck length: ${this.state.cardDeck.length}`);
   };
 
-  render() {
-    const currCardElems = this.state.currCards.map(({ name, suit }) => (
-      // Give each list element a unique key
-      <div key={`${name}${suit}`}>
-        {name} of {suit}
-      </div>
-    ));
+  decideWinner(currentScore) {
+    let decision = "";
+    if (currentScore[0] > currentScore[1]) {
```

Comment:
personally, i prefer not to use else ifs (but it's absolutely down to preference). an alternative way of writing might be:

```javascript
function decideWinner(currentScore) {
  const [firstScore, secondScore] = currentScore;
  if (firstScore > secondScore) return 'Player 1 won';
  if (secondScore > firstScore) return 'Player 2 won';
  return "It's a tie";
}
```